bindings/go/go.mod

```diff
@@ -22,11 +22,11 @@ go 1.22.4
 toolchain go1.22.5
 
 require (
+	github.com/apache/opendal-go-services/memory v0.0.0-20240719030108-74ff217cfef9
```

Comment:
Can we avoid depending `github.com/apache/opendal-go-services`?

Reply:
This is only for tests. I have separated the test into `package opendal_test` which users will not fetch when they use `go get`. Go tools will tree-shaking unnecessary dependencies.

Reply:
I understand how Go works. However, from a release standpoint, we include `opendal-go-services` in our releases even if it's just for testing. It's better to have a seperate folder or something so that we can exclude them.